exercises/connect/example.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * "Player O" plays from top to bottom, "Player X" plays from left to right.
+ * @param {string[]} initBoard - The starting state of the board
+ * @returns {Object} - The board business object
+ */
+var Board = function (initBoard) {
+  this.board = initBoard.slice();
+};
+
+Board.prototype.winner = function () {
+  var players = ['X', 'O'];
+  for (var i = 0; i < players.length; i++) {
+    if (this.checkWin(players[i])) {
+      return players[i];
+    }
+  }
+  return '';
+};
+
+Board.prototype.checkWin = function (player) {
+  var positions = this.startPositions(player);
+  for (var i = 0; i < positions.length; i++) {
+    if (this.search(positions[i], player, [])) {
+      return true;
+    }
+  }
+  return false;
+};
+
+Board.prototype.search = function (pos, XorO, isChecked) {
+  var boardInstance = this;
```

Comment:
I like this name. In other scenarios, I've seen it shorter: like `self` or `that`. What do you think of those names? They could be kind of a convention.

Reply:
This is only needed inside some functions that are nested, and thus bring their own `this`.  I tried to get rid of it entirely, but couldn't figure out how in ES5!

Let's try `self`.  I like that.

Reply:
Yeah. Fat arrow functions solve most of these problems.